src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -215,9 +212,47 @@ public long FetchOffset
             }
         }
 
+        /// <summary>
+        /// It happens when it gets consumer offset out of range exception. Try to fix the offset.
+        /// </summary>        
+        internal void ResetOffset(long resetOffset)
+        {
+            lock(this.fetchedOffsetLock)
```

Comment:
It looks like both FetchedOffset and NextRequestOffset are written by FetcherRunnable class only during running.
I think no synchronization is needed here.